apps/src/fish/FishView.jsx

```diff
@@ -53,23 +47,98 @@ class FishView extends React.Component {
     mobilePortraitWidth: PropTypes.number.isRequired
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      appWidth: $('#codeApp').width(),
+      appHeight: $('#codeApp').height()
+    };
+  }
+
   componentDidMount() {
     this.props.onMount();
+
+    this.setState({
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      appWidth: $('#codeApp').width(),
+      appHeight: $('#codeApp').height()
+    });
+
+    window.addEventListener('resize', _.debounce(this.onResize, 100));
   }
 
+  onResize = () => {
+    const windowWidth = $(window).width();
+    const windowHeight = $(window).height();
+
+    // Check that the window dimensions have actually changed to avoid
+    // unnecessary event-proessing on iOS Safari.
+    if (
+      this.state.windowWidth !== windowWidth ||
+      this.state.windowHeight !== windowHeight
+    ) {
+      const appWidth = $('#codeApp').width();
+      const appHeight = $('#codeApp').height();
+
+      this.setState({windowWidth, windowHeight, appWidth, appHeight});
+    }
+  };
+
   render() {
+    // The tutorial has a width:height ratio of 16:9.
+    const aspectRatio = 16 / 9;
+
+    let containerWidth, containerHeight;
+
+    // Constrain tutorial to 1280px maximum width.
+    const maxContainerWidth = Math.min(this.state.appWidth, 1280);
+
+    // Reducing by 27px leaves appropriate space above the small footer.
+    const maxContainerHeight =
+      Math.min(this.state.appHeight, window.innerHeight) - 27;
+
+    if (maxContainerWidth / maxContainerHeight > aspectRatio) {
+      // Constrain by height.
+      containerWidth = maxContainerHeight * aspectRatio;
+    } else if (maxContainerWidth / maxContainerWidth < aspectRatio) {
+      // Constrain by width.
+      containerWidth = maxContainerWidth;
+    }
+
+    // Constrain tutorial to 320px minimum width;
+    if (containerWidth < 320) {
```

Comment:
nit: this could be simplified to `containerWidth = Math.min(containerWidth, 320);`

Reply:
agreed, though actually `Math.max()`, right?

Reply:
woops--yes, you're right!

Reply:
I thought I'd leave the code in this function very literal, even if it's more verbose.  It felt a bit easier to return to after a year, though it's not as clever. :)